.github/workflows/performance.yml

```
@@ -0,0 +1,78 @@
+jobs:
+  # Blocking lane. Gates only on signals that are stable on shared runners:
+  # correctness under the race detector and exact allocation counts.
+  # Latency percentiles are deliberately NOT gated here, because measured spread
+  # on GitHub-hosted runners has exceeded the difference between configurations.
+  guards:
+    name: Race and allocation guards
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.12)

[error] the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)